provision/kubernetes/provisioner.go

```diff
@@ -616,6 +618,46 @@ func (p *kubernetesProvisioner) UpdateNode(opts provision.UpdateNodeOptions) err
 	return err
 }
 
+func (p *kubernetesProvisioner) Deploy(a provision.App, buildImageID string, evt *event.Event) (string, error) {
+	client, err := clusterForPool(a.GetPool())
+	if err != nil {
+		return "", err
+	}
+	newImage := buildImageID
+	if strings.HasSuffix(buildImageID, "-builder") {
+		newImage, err = image.AppNewImageName(a.GetName())
+		if err != nil {
+			return "", err
+		}
+		deployPodName, err := deployPodNameForApp(a)
+		if err != nil {
+			return "", err
+		}
+		defer cleanupPod(client, deployPodName)
+		params := createPodParams{
+			app:              a,
+			client:           client,
+			podName:          deployPodName,
+			sourceImage:      buildImageID,
+			destinationImage: newImage,
+			attachOutput:     evt,
+		}
+		err = createDeployPod(params)
+		if err != nil {
+			return "", err
+		}
+	}
+	manager := &serviceManager{
+		client: client,
+		writer: evt,
+	}
+	err = servicecommon.RunServicePipeline(manager, a, newImage, nil)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return newImage, nil
+}
+
 func (p *kubernetesProvisioner) ImageDeploy(a provision.App, imageID string, evt *event.Event) (string, error) {
```

Comment:
Can't we remove this method (along with `UploadDeploy`) now as it'll no longer be used?

Reply:
Sure! I'm just moving the tests to builder package and I will remove both methods from provider in a next commit